pkg/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -224,11 +302,21 @@ func getHugepagesAllocationUnitOptions(hugepagesSize string, hugepagesCount int3
 	}
 }
 
-func addFile(ignitionConfig *igntypes.Config, src string, dst string, mode *int) error {
-	content, err := ioutil.ReadFile(src)
-	if err != nil {
-		return err
+func getRPSUnitOptions(rpsMask string) []*unit.UnitOption {
+	cmd := fmt.Sprintf("/bin/find /sys/class/net/%%i/queues -type f -name rps_cpus -exec sh -c \"echo %s | cat > {}\" \\;", rpsMask)
+	return []*unit.UnitOption{
+		// [Unit]
+		// Description
+		unit.NewUnitOption(systemdSectionUnit, systemdDescription, "Sets network devices RPS mask"),
+		// [Service]
+		// Type
+		unit.NewUnitOption(systemdSectionService, systemdType, systemdServiceTypeOneshot),
+		// ExecStart
+		unit.NewUnitOption(systemdSectionService, systemdExecStart, cmd),
```

Comment:
Make sure the unit is executed in a deterministic order. Eg. use After and Before as needed (After=network, Before=kubelet? <-- not sure about this one).

Reply:
It does not run on system boot, the dependencies are not relevant.

Reply:
Ack, I did not know about systemd unit templates feature. A nifty trick!